### PR TITLE
ILABS-114 Swagger definition Grade/Skill/Badge

### PR DESCRIPTION
**Pull request by sam-ludlow**

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -2872,6 +3312,10 @@ components:
           type: string
           format: uuid
           example: 01234567-89ab-cdef-1234-56789abcdef0
+        has_children:
```
Comment (MireilleHelle) [inline api.swagger.yaml]: Why are we changing existing endpoints ?

Comment (ParryLuke) [reply api.swagger.yaml]: Not a breaking change, should be fine

File: api.swagger.yaml [hunk 2]
```diff
@@ -2370,6 +2370,517 @@ paths:
...
+            Location:
+              description: |
+                The URL at which the newly provisioned organisational skill may be retrieved.
+              schema:
+                type: string
+        '400':
+          description: 'The organisational skill was not saved due to malformed JSON.'
+        '403':
+          description: 'User is not authorized to create an organisational skill.'
+        '409':
```
Comment (ParryLuke) [inline api.swagger.yaml]: Just talked this through with product. Let's not add the 409 error/validation yet, sorry about that.

File: api.swagger.yaml [hunk 3]
```diff
@@ -2370,6 +2370,517 @@ paths:
...
+        '200':
+          description: Returns a single complete organisational skill object.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/organisational-skill-schema'
+        '400':
+          description: The supplied UUID was malformed.
+        '403':
+          description: 'User is not authorized to show this organisational skills.'
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: All plural issues will be addressed in the template that will be based on /organisational_skills

File: api.swagger.yaml (hunk 2 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: took the 409 out of /organisational_skills

File: api.swagger.yaml [hunk 4]
```diff
@@ -2370,6 +2370,517 @@ paths:
...
+        '422':
+          description: |
+            The organisational skill was not updated due to a Drupal validation failure or a
+            JSON schema validation failure.
+      description: Update organisational skill information.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/organisational-skill-schema'
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: PUT using create schema

File: api.swagger.yaml [hunk 5]
```diff
@@ -3714,6 +4243,129 @@ components:
...
+          format: uuid
+        published:
+          description: 'Indicates that this grade is published'
+          type: boolean
+          example: true
+    organisational-skill-schema:
+      allOf:
+        - type: object
+          properties:
+            uuid:
```
[the remark this answers is not in the document]
Comment (sam-ludlow) [reply api.swagger.yaml]: PUT using create schema

**Comment by sam-ludlow** [comment]
Closing superseded by https://github.com/invotra/api/pull/124
Decided to do all the new taxonomies in one go.

**Pull request closed without merging**